[-Dnative] JPA class Person in integration-tests/spring-data-jpa uses j.u.Random

## Describe the bug
[Person](https://github.com/quarkusio/quarkus/blob/main/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/Person.java#L30) class in integration-tests/spring-data-jpa uses `java.util.Random` as part of its implementation. When generating a native-image from that app, this fails with [graal 21.1-dev](https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/21.1.0-dev-20210323_2140) since Random is prohibited to end up in the native image heap (randomness would become predictable otherwise). The fix would be to runtime-initialize the class, but this turns out problematic since a generated JPA class, `io.quarkus.it.spring.data.jpa.PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl_Bean`, triggers initialization of the class and would, thus, need to get runtime-initialized as well. It's not clear how this could be done since the generated class name is not predictable. I'd expect for this example to be a valid use-case and as it stands there doesn't seem to be a good solution available yet to solve this runtime initialization config problem.

### Expected behavior
Have some way to allow for runtime initialized JPA-generated classes.

### Actual behavior
No way to configure runtime initialization of generated classes

## To Reproduce
1. Get a graal vm 21.1-dev build. for example https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/21.1.0-dev-20210323_2140
2. `export GRAALVM_HOME=/path/to/21.1-dev/build`
3. `./mvnw verify -Dnative -Dquarkus.native.additional-build-args="--trace-class-initialization=io.quarkus.it.spring.data.jpa.Person,--initialize-at-run-time=io.quarkus.it.spring.data.jpa.Person,--trace-object-instantiation=java.util.Random" -pl integration-tests/spring-data-jpa/`

### Configuration

Tried with that to fix the runtime init problem, but that's insufficient.

```properties
quarkus.native.additional-build-args=--initialize-at-run-time=io.quarkus.it.spring.data.jpa.Person
```

## Additional context
This fix didn't work. I guess a more elaborate, more generic fix is needed.

Example stack trace:
```
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:build (default) @ quarkus-integration-test-spring-data-jpa ---
[INFO] [org.jboss.threads] JBoss Threads version 3.2.0.Final
[INFO] [org.hibernate.Version] HHH000412: Hibernate ORM core version 5.4.30.Final
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/integration-tests/spring-data-jpa/target/quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-native-image-source-jar/quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/integration-tests/spring-data-jpa/target/quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-native-image-source-jar/quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM Version 21.1.0-dev (Mandrel Distribution) (Java Version 11.0.11-ea+2)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /disk/graal/builds/graalvm-21.1-dev-2021-03-19-ServiceLocator/bin/native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-DCoordinatorEnvironmentBean.transactionStatusManagerEnable=false -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Duser.country=US -J-Dfile.encoding=UTF-8 --trace-class-initialization=io.quarkus.it.spring.data.jpa.Person --initialize-at-run-time=io.quarkus.it.spring.data.jpa.Person --trace-object-instantiation=java.util.Random --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -jar quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:NativeLinkerOption=-no-pie -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner
[quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner:14887]    classlist:   6,819.91 ms,  1.19 GB
[quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner:14887]        (cap):     700.52 ms,  1.19 GB
[quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner:14887]        setup:   3,527.52 ms,  1.19 GB
12:10:42,891 INFO  [org.hib.Version] HHH000412: Hibernate ORM core version 999-SNAPSHOT
12:10:42,897 INFO  [org.hib.ann.com.Version] HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
12:10:42,938 INFO  [org.hib.dia.Dialect] HHH000400: Using dialect: org.hibernate.dialect.H2Dialect
[quarkus-integration-test-spring-data-jpa-999-SNAPSHOT-runner:14887]     analysis:  15,206.13 ms,  2.00 GB
Error: Classes that should be initialized at run time got initialized during image building:
 io.quarkus.it.spring.data.jpa.Person the class was requested to be initialized at run time (from the command line). io.quarkus.runner.ApplicationImpl caused initialization of this class with the following trace: 
	at io.quarkus.it.spring.data.jpa.Person.<clinit>(Person.java)
	at java.lang.Class.forName0(Unknown Source)
	at java.lang.Class.forName(Class.java:398)
	at io.quarkus.it.spring.data.jpa.PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl_Bean.<init>(PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl_Bean.zig:171)
	at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans2(Default_ComponentsProvider.zig:646)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:43)
	at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:112)
	at io.quarkus.arc.Arc.initialize(Arc.java:20)
	at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:40)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy_0(ArcProcessor$generateResources-1025303321.zig:76)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy(ArcProcessor$generateResources-1025303321.zig:40)
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:240)


com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 io.quarkus.it.spring.data.jpa.Person the class was requested to be initialized at run time (from the command line). io.quarkus.runner.ApplicationImpl caused initialization of this class with the following trace: 
	at io.quarkus.it.spring.data.jpa.Person.<clinit>(Person.java)
	at java.lang.Class.forName0(Unknown Source)
	at java.lang.Class.forName(Class.java:398)
	at io.quarkus.it.spring.data.jpa.PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl_Bean.<init>(PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl_Bean.zig:171)
	at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans2(Default_ComponentsProvider.zig:646)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:43)
	at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:112)
	at io.quarkus.arc.Arc.initialize(Arc.java:20)
	at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:40)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy_0(ArcProcessor$generateResources-1025303321.zig:76)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy(ArcProcessor$generateResources-1025303321.zig:40)
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:240)


	at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.checkDelayedInitialization(ConfigurableClassInitialization.java:545)
	at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.duringAnalysis(ClassInitializationFeature.java:228)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$14(NativeImageGenerator.java:763)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:763)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:580)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$2(NativeImageGenerator.java:493)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  58.675 s
[INFO] Finished at: 2021-03-25T12:10:57+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project quarkus-integration-test-spring-data-jpa: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:209)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.RuntimeException: Image generation failed. Exit code: 1
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:350)
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:181)
[ERROR] 	... 10 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/DataGenerator.java
+++ b/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/DataGenerator.java
@@ -1,0 +1,25 @@
+package io.quarkus.it.spring.data.jpa;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Random;
+
+public class DataGenerator {
+
+    public static final Random RANDOM = new Random();
+
+    public static Date randomDate() {
+        return Date.from(LocalDate.now().minusDays(RANDOM.nextInt(3000)).atStartOfDay()
+                .atZone(ZoneId.systemDefault())
+                .toInstant());
+    }
+
+    public static int randomAge() {
+        return RANDOM.nextInt(100);
+    }
+
+    public static boolean randomBoolean() {
+        return RANDOM.nextBoolean();
+    }
+}

--- a/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/Person.java
+++ b/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/Person.java
@@ -1,11 +1,8 @@
 package io.quarkus.it.spring.data.jpa;
 
-import java.time.LocalDate;
-import java.time.ZoneId;
 import java.util.Date;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Random;
 import java.util.Set;
 
 import javax.json.bind.annotation.JsonbDateFormat;
@@ -26,8 +23,6 @@ import javax.persistence.Table;
 
 @Entity
 public class Person {
-
-    private static final Random RANDOM = new Random();
 
     @Id
     @SequenceGenerator(name = "personSeqGen", sequenceName = "personSeq", initialValue = 100, allocationSize = 1)
@@ -54,11 +49,9 @@ public class Person {
 
     public Person(String name) {
         this.name = name;
-        this.age = RANDOM.nextInt(100);
-        this.joined = Date.from(LocalDate.now().minusDays(RANDOM.nextInt(3000)).atStartOfDay()
-                .atZone(ZoneId.systemDefault())
-                .toInstant());
-        this.active = RANDOM.nextBoolean();
+        this.age = DataGenerator.randomAge();
+        this.joined = DataGenerator.randomDate();
+        this.active = DataGenerator.randomBoolean();
     }
 
     public Person() {

--- a/integration-tests/spring-data-jpa/src/main/resources/application.properties
+++ b/integration-tests/spring-data-jpa/src/main/resources/application.properties
@@ -5,6 +5,10 @@ quarkus.datasource.jdbc.max-size=8
 
 quarkus.hibernate-orm.dialect=org.hibernate.dialect.H2Dialect
 quarkus.hibernate-orm.database.generation=drop-and-create
+# DataGenerator uses java.util.Random => needs to be runtime initialized
+# as j.u.Random is prohibited in the native image heap. Otherwise
+# random data becomes deterministic
+quarkus.native.additional-build-args=--initialize-at-run-time=io.quarkus.it.spring.data.jpa.DataGenerator
 #quarkus.hibernate-orm.log.sql=true
 
 %prod.quarkus.hibernate-orm.sql-load-script=import.sql

